text/chapter5.md

```diff
@@ -360,31 +339,14 @@ Algebraic data types provide a type-safe way to solve this sort of problem, if t
 Here is how `Shape` might be represented as an algebraic data type:
 
 ```haskell
-data Shape
-  = Circle Point Number
-  | Rectangle Point Number Number
-  | Line Point Point
-  | Text Point String
-```
-
-The `Point` type might also be defined as an algebraic data type, as follows:
+{{#include ../exercises/chapter5/src/Data/Picture.purs:Shape}}
 
-```haskell
-data Point = Point
-  { x :: Number
-  , y :: Number
-  }
+{{#include ../exercises/chapter5/src/Data/Picture.purs:Point}}
 ```
 
-The `Point` data type illustrates some interesting points:
-
-- The data carried by an ADT's constructors doesn't have to be restricted to primitive types: constructors can include records, arrays, or even other ADTs.
-- Even though ADTs are useful for describing data with multiple constructors, they can also be useful when there is only a single constructor.
-- The constructors of an algebraic data type might have the same name as the ADT itself. This is quite common, and it is important not to confuse the `Point` _type constructor_ with the `Point` _data constructor_ - they live in different namespaces.
```

Comment:
These points are absorbed the newtype section. I don't think they need to be duplicated for ADTs too.
Historical question, was this section written before `newtype`s existed?